### ISSUE TITLE
Configuration Error in v2.2.1

### ISSUE THREAD

**Issue by TBSliver** [grafana/loki#3821]
**Describe the bug**

Tried to start Loki with example configuration plus new retention options for compactor, got an error that it didnt recognise any of the options:

```
Jun 07 17:55:13 redacted.domain.name systemd[1]: Started Loki: Like Prometheus, but for logs..
Jun 07 17:55:13 redacted.domain.name loki-linux-amd64[460896]: failed parsing config: /etc/loki/config.yml: yaml: unmarshal errors:
Jun 07 17:55:13 redacted.domain.name loki-linux-amd64[460896]:   line 50: field retention_enabled not found in type compactor.Config
Jun 07 17:55:13 redacted.domain.name loki-linux-amd64[460896]:   line 51: field retention_delete_delay not found in type compactor.Config
Jun 07 17:55:13 redacted.domain.name loki-linux-amd64[460896]:   line 52: field retention_delete_worker_count not found in type compactor.Config
Jun 07 17:55:13 redacted.domain.name loki-linux-amd64[460896]:   line 57: field retention_period not found in type validation.plain
Jun 07 17:55:13 redacted.domain.name systemd[1]: loki.service: Main process exited, code=exited, status=1/FAILURE
```

**To Reproduce**

Start Loki with the following config:

```yaml
auth_enabled: false

server:
  http_listen_address: 127.0.0.1
  http_listen_port: 3100
  grpc_listen_address: 127.0.0.1
  grpc_listen_port: 9096
  log_level: warn

ingester:
  wal:
    enabled: true
    dir: /opt/loki/wal
  lifecycler:
    address: 127.0.0.1
    ring:
      kvstore:
        store: inmemory
      replication_factor: 1
    final_sleep: 0s
  chunk_idle_period: 1h       # Any chunk not receiving new logs in this time will be flushed
  max_chunk_age: 1h           # All chunks will be flushed when they hit this age, default is 1h
  chunk_target_size: 1048576  # Loki will attempt to build chunks up to 1.5MB, flushing first if chunk_idle_period or max_chunk_age is reached first
  chunk_retain_period: 30s    # Must be greater than index read cache TTL if using an index cache (Default index read cache TTL is 5m)
  max_transfer_retries: 0     # Chunk transfers disabled

schema_config:
  configs:
    - from: 2020-10-24
      store: boltdb-shipper
      object_store: filesystem
      schema: v11
      index:
        prefix: index_
        period: 24h

storage_config:
  boltdb_shipper:
    active_index_directory: /opt/loki/store/boltdb-shipper-active
    cache_location: /opt/loki/store/boltdb-shipper-cache
    cache_ttl: 24h         # Can be increased for faster performance over longer query periods, uses more disk space
    shared_store: filesystem
  filesystem:
    directory: /opt/loki/store/chunks

compactor:
  working_directory: /opt/loki/store/boltdb-shipper-compactor
  shared_store: filesystem
  compaction_interval: 10m
  retention_enabled: true
  retention_delete_delay: 2h
  retention_delete_worker_count: 5

limits_config:
  reject_old_samples: true
  reject_old_samples_max_age: 168h
  retention_period: 744h # 31 days

chunk_store_config:
  max_look_back_period: 0s

table_manager:
  retention_deletes_enabled: false
  retention_period: 0s

ruler:
  storage:
    type: local
    local:
      directory: /opt/loki/store/rules
  rule_path: /opt/loki/store/rules-temp
  alertmanager_url: http://localhost:9093
  ring:
    kvstore:
      store: inmemory
  enable_api: true
```

Gives the error shown above. Only major change from the default one is using the retention settings as specified in the documentation [found here](https://grafana.com/docs/loki/latest/operations/storage/retention/#Compactor)

**Expected behavior**

Start with no errors, with retention policy intact.

**Environment:**

- Loki version 2.2.1 (latest release)
- Bare Metal VM running Icinga2 and Grafana alongside (monitoring box, of course)
- Ubuntu 20.04 LTS

**Screenshots, Promtail config, or terminal output**

See above for config etc.

**Comment by TBSliver** [comment]
Note that for now Im using the 'older' `table_manager` retention config, but as the documentation says that is going to be phased out so wanted to go for the newer system so less wrangling in the future!

**Comment by dannykopping** [comment]
Unfortunately, retention is not available yet in v2.2.1.
It is merged into `main` but will only be released in v2.3 (coming soon).

**Comment by dannykopping** [comment]
@cyriltovena @KMiller-Grafana we should probably specify in the [docs](https://grafana.com/docs/loki/latest/operations/storage/retention/#Compactor) that this is not yet available
